server/graphql/v2/object/Host.js

```diff
@@ -437,6 +464,148 @@ export const Host = new GraphQLObjectType({
           };
         },
       },
+      contributionStats: {
+        type: new GraphQLNonNull(ContributionStats),
+        args: {
+          account: {
+            type: new GraphQLList(new GraphQLNonNull(AccountReferenceInput)),
+            description: 'A collection of accounts for which the contribution stats should be returned.',
+          },
+          dateFrom: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics beginning from this date.',
+          },
+          dateTo: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics until this date.',
+          },
+        },
+        async resolve(host, args, req) {
+          if (!req.remoteUser?.isAdmin(host.id)) {
+            throw new Unauthorized('You need to be logged in as an admin of the host to see the contribution stats.');
+          }
+          const where = {
+            HostCollectiveId: host.id,
+            kind: TransactionKind.CONTRIBUTION,
+            type: TransactionTypes.CREDIT,
+            isRefund: false,
+            RefundTransactionId: null,
+          };
+          const numberOfDays = getNumberOfDays(args.dateFrom, args.dateTo, host);
+          const dateRange = getFilterDateRange(args.dateFrom, args.dateTo);
+          if (dateRange) {
+            where.createdAt = dateRange;
+          }
+          if (args.account) {
+            const collectives = await fetchAccountsWithReferences(args.account, { throwIfMissing: true });
+            const collectiveIds = collectives.map(collective => collective.id);
+            where.CollectiveId = { [Op.in]: collectiveIds };
+          }
+          const contributionsCount = await models.Transaction.count({
+            where,
+          });
+
+          const oneTimeContributionCountByOrderId = await models.Transaction.count({
+            where,
+            include: [{ model: models.Order, where: { interval: null } }],
+            group: ['OrderId'],
+          });
+          const oneTimeContributionsCount = oneTimeContributionCountByOrderId.length;
```

Comment:
It's not returning a number when we group it by `OrderId`. What it returns is an array with one time contribution counts by order id (hence the name `oneTimeContributionCountByOrderId`). Something like,

`[{orderId: 10882, count: 2}, {orderId: 10883, count: 5}, ...]`

Reply:
Then it sounds like this query is far from being optimal. For large collectives, it's going to return an array with 5000+ objects while we're only interested in one integer value.

In raw SQL, we could write something like:
```sql
SELECT COUNT (DISTINCT "OrderId") FROM "Transactions" t 
```

It would be great if we could something like that with Sequelize. Otherwise we can still fallback on raw SQL.

Reply:
@Betree : I should have caught this in the beginning. We could use the `distinct` parameter; https://sequelize.org/master/class/lib/model.js~Model.html#static-method-count. I've changed that and now it should look better. 😉